src/iiif/image_handling.py

```diff
@@ -123,3 +126,96 @@ def scale_image(content, source_file, scaling, content_type):
     scaled_image_data = image_stream.getvalue()
 
     return scaled_image_data
+
+# TODO: Support percentage string: pct:x,y,w,h
+def parse_region_string(region):
+    """
+    Parse the region string from the url (either 'full' or 'x,y,w,h' in pixels)
+
+    :param region: The region string from the url
+    :return: Tuple containing the desired x, y, width and height.
+    """
+    try:
+        parts = region.split(",")
+        if len(parts) != 4:
+            raise ValueError("Invalid format for region")
+        
+        if not (parts[0] or parts[1] or parts[2] or parts[3]):
+            raise ValueError('Invalid format for region')
+
+        desired_x = int(parts[0])
+        desired_y = int(parts[1])
+        desired_width = int(parts[2])
+        desired_height = int(parts[3])
+    except ValueError:
+        raise tools.ImmediateHttpResponse(
+            response=HttpResponse(MALFORMED_REGION_PARAMETER, status=400)
+        )
+
+    return desired_x, desired_y, desired_width, desired_height
+
+# TODO: Move to an util file
+def clamp(n, minn, maxn):
+    return max(min(maxn, n), minn)
+
+# TODO: Extract sub functions to own functions for:
+# - asserting the region is valid
+# - return original when no crop was applied
```

Comment:
Deze zit er nu al in toch?

Reply:
I would love to make the function more clear. We can do that by abstracting these mathy parts to their own (pure) subfunctions. In this way, this main function will become a lot more clear in what it tries to do without all the math cluttering it.